From 3b6cdfa4d005d3c3617e3055bb5dcfe8fb2ff917 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Fri, 1 Jan 2021 14:19:59 +0100
Subject: [PATCH] luci-base.json: add path to /bin/nslookup

Related to: https://gitlab.nic.cz/turris/turris-build/-/blob/v5.1.4/patches/luci/hack/0001-luci-base-add-path-for-busybox-ntpd.patch

We are installing busybox to /{s,}bin instead of using /usr/bin as
default in OpenWrt and LuCI does not count with that as it is using hard
coded paths and it results that in Diagnostics in LuCI it shows
PermissionError: Permission denied while using nslookup.

Ping and traceroute there counts with two paths:
/bin and /usr/bin

Fixes:
- https://forum.turris.cz/t/luci-permission-error-on-nslookup/14540
- https://github.com/openwrt/luci/issues/4705
---
 modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json b/modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json
index 8b8481b1c..4ffc7d241 100644
--- a/modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json
+++ b/modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json
@@ -44,6 +44,7 @@
 				"/bin/ping6 *": [ "exec" ],
 				"/bin/traceroute *": [ "exec" ],
 				"/bin/traceroute6 *": [ "exec" ],
+				"/bin/nslookup *": [ "exec" ],
 				"/sbin/ip -4 neigh show": [ "exec" ],
 				"/sbin/ip -4 route show table all": [ "exec" ],
 				"/sbin/ip -6 neigh show": [ "exec" ],
-- 
2.25.1

